app/decorators/membership_view_model_decorator.rb

```diff
@@ -25,6 +25,17 @@ def join_group_button
     end
   end
 
+  def withdraw_membership_button
+    group = membership_view_model.group
+    membership = current_user.membership_for(group)
```

Comment:
Maybe we should move these two assignments to private or helper methods. It could improve the readability of the whole file.

Reply:
Or just the first because the membership is only used in this method.

Reply:
For the moment I'd leave like this, it's better not to mix refactoring and new features in the same PR.

If you take into account that every other method is called on `membership_view_model` this also fits better, though I can see potential in a simple delegation like `delegate :group, to: :membership_view_model`